physics/kepler_orbit_body.hpp

```diff
@@ -157,94 +213,478 @@ KeplerOrbit<Frame>::KeplerOrbit(
+      hyperbolic_excess_velocity = Cbrt(2 * π * Sqrt(-Pow<2>(μ / T)));
+    } else if (hyperbolic_mean_motion) {
+      AngularFrequency const& n_over_i = *hyperbolic_mean_motion;
+      semimajor_axis = -Cbrt(μ / Pow<2>(n_over_i / Radian));
+      specific_energy = Pow<2>(Cbrt(μ * n_over_i / Radian)) / 2;
+      characteristic_energy =  Pow<2>(Cbrt(μ * n_over_i / Radian));
+      period = 2 * π * Radian / Sqrt(-Pow<2>(n_over_i));
+      mean_motion = Sqrt(-Pow<2>(n_over_i));
+      hyperbolic_excess_velocity = Sqrt(Pow<2>(Cbrt(μ * n_over_i / Radian)));
+    } else if (hyperbolic_excess_velocity) {
+      Speed const& v_inf = *hyperbolic_excess_velocity;
+      semimajor_axis = -μ / Pow<2>(v_inf);
+      specific_energy = Pow<2>(v_inf) / 2;
+      characteristic_energy =  Pow<2>(v_inf);
+      period = 2 * π * Sqrt(-Pow<2>(μ) / Pow<6>(v_inf));
+      mean_motion = Sqrt(-Pow<6>(v_inf) / Pow<2>(μ)) * Radian;
+      hyperbolic_mean_motion = Sqrt(Pow<6>(v_inf) / Pow<2>(μ)) * Radian;
+    }
+  }
+  if (must_complete_semiminor_axis) {
+    if (semiminor_axis) {
+      impact_parameter = Sqrt(-Pow<2>(*semiminor_axis));
+    } else if (impact_parameter) {
+      semiminor_axis = Sqrt(-Pow<2>(*impact_parameter));
+    }
+  }
+  if (must_complete_semilatus_rectum) {
+    if (semilatus_rectum) {
+      specific_angular_momentum = Sqrt(μ * *semilatus_rectum) / Radian;
+    } else if (specific_angular_momentum) {
+      SpecificAngularMomentum const& h = *specific_angular_momentum;
+      semilatus_rectum = Pow<2>(h * Radian) / μ;
+    }
+  }
+}
+
+template<typename Frame>
+void KeplerOrbit<Frame>::CompleteConicParameters(
+    KeplerianElements<Frame>& elements,
+    GravitationalParameter const& μ) {
+  auto& eccentricity = elements.eccentricity;
+  auto& asymptotic_true_anomaly = elements.asymptotic_true_anomaly;
+  auto& turning_angle = elements.turning_angle;
+  auto& semimajor_axis = elements.semimajor_axis;
+  auto& specific_energy = elements.specific_energy;
+  auto& characteristic_energy = elements.characteristic_energy;
+  auto& mean_motion = elements.mean_motion;
+  auto& period = elements.period;
+  auto& hyperbolic_mean_motion = elements.hyperbolic_mean_motion;
+  auto& hyperbolic_excess_velocity = elements.hyperbolic_excess_velocity;
+  auto& semiminor_axis = elements.semiminor_axis;
+  auto& impact_parameter = elements.impact_parameter;
+  auto& semilatus_rectum = elements.semilatus_rectum;
+  auto& specific_angular_momentum = elements.specific_angular_momentum;
+  auto& periapsis_distance = elements.periapsis_distance;
+  auto& apoapsis_distance = elements.apoapsis_distance;
+
+  CompleteConicParametersByCategory(elements, μ);
+
+  CHECK_EQ(eccentricity.has_value() + semimajor_axis.has_value() +
+               semiminor_axis.has_value() + semilatus_rectum.has_value() +
+               periapsis_distance.has_value() + apoapsis_distance.has_value(),
+           2) << elements;
```

Comment:
As discussed, no (and we should eventually clean things up to consistently put expected values last).